packages/side-nav/src/SideNavItem.tsx

```diff
@@ -191,23 +173,28 @@ const SideNavItem: ExtendableBox<
     ariaCurrentValue = AriaCurrentValue.Page,
     className,
     children,
+    onClick: onClickProp,
+    glyph,
+    ...rest
   } = props;
+  const { usingKeyboard: showFocus } = useUsingKeyboardContext();
 
-  const rest = omit(props, [
-    'active',
-    'disabled',
-    'ariaCurrentValue',
-    'className',
-    'children',
-  ]);
+  const onClick = disabled
+    ? (e: React.MouseEvent) => {
+        e.nativeEvent.stopImmediatePropagation();
+        e.preventDefault();
+      }
+    : onClickProp;
```

Comment:
I think what I meant was, wouldn't we not want to call this onClick prop when disabled. Instead const onClick = !disabled && () => {}

Reply:
If this is a link and we didn't pass an onClick when the item is disabled, clicking on the link would still navigate you. That's why we still have an onClick handler when it's disabled.